client/app/project/project.controller.js

```diff
@@ -7,37 +7,98 @@
      */
     angular
         .module('taskingManager')
-        .controller('projectController', ['$location', 'mapService', 'projectService', 'styleService', projectController]);
+        .controller('projectController', ['$scope', '$location', 'mapService', 'projectService', 'styleService', projectController]);
 
-    function projectController($location, mapService, projectService, styleService) {
+    function projectController($scope, $location, mapService, projectService, styleService) {
         var vm = this;
         vm.project = null;
         vm.map = null;
         vm.currentTab = '';
+        vm.mappingStep = '';
+
+        //selected task
+        vm.selectedTask = null;
+        vm.isSelectTaskMappable = false;
+
+        //interaction
+        var select = new ol.interaction.Select({
+            style: styleService.getSelectedStyleFunction
+        });
 
         activate();
 
         function activate() {
             //TODO: Set up sidebar tabs
             vm.currentTab = 'description';
+            vm.mappingStep = 'select';
             mapService.createOSMMap('map');
             vm.map = mapService.getOSMMap();
+
+            vm.map.addInteraction(select);
```

Comment:
Only add interaction when it hasn't been defined before. Every time the user lands on the /project page, another interaction is added to the map.